src/main/java/edu/wpi/grip/core/sources/WebCamSource.java

```diff
@@ -0,0 +1,188 @@
+package edu.wpi.grip.core.sources;
+
+
+import com.google.common.eventbus.EventBus;
+import edu.wpi.grip.core.OutputSocket;
+import edu.wpi.grip.core.SocketHint;
+import edu.wpi.grip.core.Source;
+import org.bytedeco.javacpp.opencv_core.Mat;
+import org.bytedeco.javacpp.opencv_videoio;
+import org.bytedeco.javacv.Frame;
+import org.bytedeco.javacv.FrameGrabber;
+import org.bytedeco.javacv.OpenCVFrameConverter;
+import org.bytedeco.javacv.OpenCVFrameGrabber;
+
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class WebCamSource implements Source {
+
+    public enum SourceType {
```

Comment:
This really isn't necessary, is it?  It looks like you're just copying a JavaCV enum.  Now we have to make sure that our copy always stays in sync with the JavaCV one.